### PR TITLE
go: add vpnkit-expose-port-controller for Kubernetes

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

**Comment by errordeveloper** [comment]
As previously seen in https://github.com/linuxkit/linuxkit/pull/2670.

File: go/cmd/vpnkit-expose-port-controller/vpnkit-expose-port-controller.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+          labels:
+            name: vpnkit-expose-port-controller
+        spec:
+          serviceAccount: vpnkit-expose-port-controller
+          tolerations:
+            - effect: NoSchedule
+              operator: Exists
+          containers:
+            - name: app
+              image: 'docker/kubernetes-vpnkit-expose-port-controller:current'
```
Comment (justincormack) [inline go/cmd/vpnkit-expose-port-controller/vpnkit-expose-port-controller.yaml]: thats not actually the image...

Comment (errordeveloper) [reply go/cmd/vpnkit-expose-port-controller/vpnkit-expose-port-controller.yaml]: What should we call this image?

Comment (justincormack) [reply go/cmd/vpnkit-expose-port-controller/vpnkit-expose-port-controller.yaml]: I think we should put it under `linuxkitprojects` for now

File: go/cmd/vpnkit-expose-port-controller/Makefile [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+HASH?=$(shell git ls-tree HEAD -- ../$(notdir $(CURDIR)) | awk '{print $$3}')
+
+.PHONY: build
+build:
+	docker build -t $(ORG)/$(IMAGE):$(HASH) .
+
+.PHONY: run
+run: build
+	docker save $(ORG)/$(IMAGE):$(HASH) | docker -H unix://../build/kubernetes-for-mac-efi-state/guest.00000948 load
+	docker -H unix://../build/kubernetes-for-mac-efi-state/guest.00000948 tag $(ORG)/$(IMAGE):$(HASH) $(ORG)/$(IMAGE):current
```
Comment (errordeveloper) [inline go/cmd/vpnkit-expose-port-controller/Makefile]: Just realised that I forgot to remove these two lines. Let's decide on the image and I'll update the PR.

File: go/cmd/vpnkit-expose-port-controller/main.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+)
+
+// TODO cleanup logging, i.e. use debug log level for commands
+// TODO implement updatePorts
+// TODO emit Kubernetes events on errors (and perhaps success), so they can be observed by the user (via `kubectl get events`
+//      or `kubectl describe service`), make sure errors user sees are meaningful
+// TODO consider controller pod lifetime and lifetime of children, there are a few ways to go about it, do we really need to
+//      write PID files? we want to persist them, we should mount state dir or use ports dir? or we could just use a channel,
+//      if we expect children to die together with the parent controller...
+
```
Comment (errordeveloper) [inline go/cmd/vpnkit-expose-port-controller/main.go]: Happy create issues for these TODOs.

**Pull request merged**